nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -535,7 +540,8 @@ in
         ''-append "$(cat ${config.system.build.toplevel}/kernel-params) init=${config.system.build.toplevel}/init regInfo=${regInfo}/registration ${consoles} $QEMU_KERNEL_PARAMS"''
       ])
       (mkIf cfg.useEFIBoot [
-        "-pflash $TMPDIR/bios.bin"
+        "-drive if=pflash,format=raw,unit=0,readonly,file=${efiFirmware}"
+        "-drive if=pflash,format=raw,unit=1,file=$TMPDIR/EFI_VARS.fd"
```

Comment:
This should probably be placed in the cwd like nixos.qcow2, since it's part of the machine's state.

Reply:
Great point! I've added a new commit which treats this as state like the disk image.   Matching `virtualisation.diskImage`, I've made a new option `virtualisation.efiVars`, which defaults to `./${vmName}-efi-vars.fd`.  This way, efi vars across multiple machines that have different hostnames would not overwrite each other.